api/src/main/java/org/regenstrief/linkage/io/OpenMRSReader.java

```diff
@@ -123,15 +129,38 @@ private Criteria createCriteria() {
 	}
 	
 	private DbSession createHibernateSession() {
-		HibernateConnection connection = new HibernateConnection();
-		return connection.getSessionFactory().getCurrentSession();
+		return Context.getRegisteredComponent("dbSessionFactory", DbSessionFactory.class).getCurrentSession();
 	}
 	
 	private void updatePatientList() {
 		List<?> list;
 		try {
 			resultMode = MODE_PATIENT;
-			list = createCriteria().list();
+			List<?> results = createCriteria().list();
```

Comment:
Have the changes in this method anything to do with voided data?

Reply:
Yes, because now that we perform a left outer join instead of a full join we get back duplicate rows so we need to loop through them to only return unique patient records.

Reply:
Any performance implication of this?

Reply:
There is no way you can enforce the uniqueness with the join because that's not the job of a join, this is one of those cases where there is no way around it, we have to perform a left join because we don't want to miss some patient records because they have no address or person attribute entry. The duplicates are because we are performing left joins so we MUST filter them out and we can't avoid a left join, hibernate internally does the same and filters our duplicates internally in memory, so this is not something new.

Reply:
But wait to merge this